### PR TITLE
Corrections to the README file

### LINKED ISSUE #47: README errors
[This code snippet](https://github.com/ORNL/ReSolve/blob/develop/README.md?plain=1#L13) from the README is incorrect.

```shell
$ git clone https://code.ornl.gov/peles/resolve.git
$ mkdir build && cd build
$ cmake ../resolve
$ make
```
=>

1. wrong clone url
2. cmake .. (not ../resolve; that causes a CMAKE_MACOSX_RPATH issue)

### PR DESCRIPTION
**Pull request by pelesh**
- Updated install and test instructions. Fixes #47 
- Fixed some typos.
- Explicitly states that ReSolve's license does not cover its dependencies.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,15 +5,15 @@ ReSolve is a library of GPU-resident linear solver. It contains iterative and di
...
 - CUDA >= 11.4
 - CMake >= 3.22
 
 To build it:
 ```shell
-$ git clone https://code.ornl.gov/peles/resolve.git
+$ git clone https://github.com/ORNL/ReSolve.git
 $ mkdir build && cd build
-$ cmake ../resolve
+$ cmake ../ReSolve
```
Comment (jaelynlitz) [inline README.md:16]: This still doesn't quite work, as the cmake command is running from inside the build directory (line 15). So if you are running from the build dir (`ReSolve/build`), syntax is typically `cmake .` 
```suggestion
$ cmake .
```

File: README.md [hunk 2]
```diff
@@ -5,15 +5,15 @@ ReSolve is a library of GPU-resident linear solver. It contains iterative and di
 ## Getting started
 
 Dependencies:
-- KLU, AMD and COLAMD libraries from SuiteSparse
+- KLU, AMD and COLAMD libraries from SuiteSparse >= 5.0
 - CUDA >= 11.4
```
Comment (kswirydo) [inline README.md:9]: Now you can buil with or without cuda. Also can use hip not cuda. So this should be reworded.

**Review by kswirydo — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (pelesh) [reply README.md]: Updated per your suggestion.

**Review by kswirydo — APPROVED** [review]
okay now it looks good to me.

**Review by jaelynlitz — APPROVED** [review]

**Pull request merged**